app/uk/gov/hmrc/merchandiseinbaggagefrontend/controllers/ReviewGoodsController.scala

```diff
@@ -39,15 +44,27 @@ class ReviewGoodsController @Inject()(override val controllerComponents: Message
     }
   }
 
-  val onSubmit: Action[AnyContent] = actionProvider.journeyAction { implicit request =>
-    request.declarationJourney.goodsEntries.declarationGoodsIfComplete.fold(actionProvider.invalidRequest) { goods =>
+  val onSubmit: Action[AnyContent] = actionProvider.journeyAction.async { implicit request =>
+    request.declarationJourney.goodsEntries.declarationGoodsIfComplete.fold(actionProvider.invalidRequestF) { goods =>
       form
         .bindFromRequest()
         .fold(
-          formWithErrors => BadRequest(view(formWithErrors, goods)),
+          formWithErrors => Future.successful(BadRequest(view(formWithErrors, goods))),
           declareMoreGoods =>
-            if (declareMoreGoods) Redirect(routes.SearchGoodsController.onPageLoad())
-            else Redirect(routes.SkeletonJourneyController.taxCalculation())
+            if (declareMoreGoods) {
+              val updatedGoodsEntries = request.declarationJourney.goodsEntries.entries :+ GoodsEntry.empty
```

Comment:
we could probably add an operation to the GoodsEntries object which would add GoodsEntry.empty to a given GoodsEntries